staging/src/k8s.io/component-helpers/scheduling/corev1/nodeaffinity/nodeaffinity_test.go

```diff
@@ -218,8 +218,8 @@ func TestPreferredSchedulingTermsScore(t *testing.T) {
 				},
 				&field.Error{
 					Type:   field.ErrorTypeInvalid,
-					Field:  "[2].matchExpressions[0]",
-					Detail: `invalid label key "invalid key": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')`,
+					Field:  "[2].matchExpressions[0].key",
```

Comment:
If there are multiple keys (multiple expressions), how does the user know which one this message is about?

Reply:
Output contains field. It would like `[[2].matchExpressions[0].key: Invalid value: REASON, [2].matchExpressions[1].key: Invalid value: REASON]`
https://github.com/kubernetes/kubernetes/blob/65395137b5c4354893408dcc8d078954210ffeff/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go#L41-L43

Reply:
I'm really sorry to be so dense, but I've found that's often much more accurate to look at the actually rendered strings than mentally follow the string rendering code. Can you make a test that prints out the rendered error messages from the invalid cases and run it with and without your change here, so we can compare the difference? Or just manually put some invalid selectors through the API.

In a change like this I really want to be sure that the error messages are helping users more and not just making the code more regular.

Reply:
It would like:
![image](https://user-images.githubusercontent.com/14567045/104431842-53551280-55c3-11eb-90e0-95cb5a1b6dd3.png)

test yaml:
<details>

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: with-node-affinity
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchFields:
          - key: "metadata.name"
            operator: In
            values:
            - a
            - b
        - matchExpressions:
          - key: "invalid key"
            operator: In
            values:
            - b
            - c
  containers:
  - name: with-node-affinity
    image: k8s.gcr.io/pause:3.2
```

</details>

Reply:
`kube-scheduler` has a similar output, just not pretty print like `kubectl`, because `kubectl` formats a http response, `kube-scheduler` simply call `.Error()`.
![image](https://user-images.githubusercontent.com/14567045/104444662-0ed17300-55d3-11eb-9640-4528974f82f7.png)

Reply:
This PR doesn't introduce this error format, it just makes `labels.NewRequirement` return all parsing errors.

Example (without/with this patch):
![image](https://user-images.githubusercontent.com/14567045/104446742-dd0ddb80-55d5-11eb-91be-6149abc7b40e.png)

<details>

```yaml
apiVersion: kubescheduler.config.k8s.io/v1beta1
kind: KubeSchedulerConfiguration
profiles:
  - schedulerName: default-scheduler
  - schedulerName: test-err-scheduler
    pluginConfig:
    - name: NodeAffinity
      args:
        addedAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
            - matchExpressions:
              - key: "invalid key"
                operator: Gt
                values:
                - b
                - c
clientConnection:
  kubeconfig: kubeconfig
```

</details>

Reply:
OK, thanks for humoring me :)

...I wouldn't say this is amazingly clear, but it's maybe slightly better than it was, and I think people will be able to figure it out.